community/io/src/main/java/org/neo4j/io/fs/DefaultFileSystemAbstraction.java

```diff
@@ -133,8 +135,13 @@ public void deleteRecursively( File directory ) throws IOException
     }
 
     @Override
-    public boolean renameFile( File from, File to ) throws IOException
+    public boolean move( File from, File to, CopyOption... copyOptions ) throws IOException
     {
+        if ( copyOptions.length > 0 )
+        {
+            Files.move( from.toPath(), to.toPath(), copyOptions );
+            return true; // will throw if failure
+        }
         return FileUtils.renameFile( from, to );
```

Comment:
This is pretty ugly. The behaviour, potential exceptions and side-effects change depending on whether there are copy options or not, even if they otherwise should make no functional difference.

Have you tried using `Files.move` always, and checked if it works or breaks on all relevant platforms?